src/test/java/seedu/task/storage/JsonAdaptedTaskTest.java

```diff
@@ -35,39 +40,73 @@ public void toModelType_validTaskDetails_returnsTask() throws Exception {
+        JsonAdaptedTask task =
+                new JsonAdaptedTask(null, VALID_DESCRIPTION, VALID_TAGS, EMPTY_DATE, EMPTY_DATE, EMPTY_DATE);
         String expectedMessage = String.format(MISSING_FIELD_MESSAGE_FORMAT, Name.class.getSimpleName());
         assertThrows(IllegalValueException.class, expectedMessage, task::toModelType);
     }
 
     @Test
     public void toModelType_invalidDescription_throwsIllegalValueException() {
         JsonAdaptedTask task =
-                new JsonAdaptedTask(VALID_NAME, INVALID_DESCRIPTION, VALID_TAGS);
+                new JsonAdaptedTask(VALID_NAME, INVALID_DESCRIPTION, VALID_TAGS, EMPTY_DATE, EMPTY_DATE, EMPTY_DATE);
         String expectedMessage = Description.MESSAGE_CONSTRAINTS;
         assertThrows(IllegalValueException.class, expectedMessage, task::toModelType);
     }
 
     @Test
     public void toModelType_nullDescription_throwsIllegalValueException() {
-        JsonAdaptedTask task = new JsonAdaptedTask(VALID_NAME, null, VALID_TAGS);
+        JsonAdaptedTask task =
+                new JsonAdaptedTask(VALID_NAME, null, VALID_TAGS, EMPTY_DATE, EMPTY_DATE, EMPTY_DATE);
         String expectedMessage = String.format(MISSING_FIELD_MESSAGE_FORMAT, Description.class.getSimpleName());
         assertThrows(IllegalValueException.class, expectedMessage, task::toModelType);
     }
 
+    @Test
+    public void toModelType_invalidDeadline_throwsIllegalValueException() {
+        JsonAdaptedTask task =
+                new JsonAdaptedTask(VALID_NAME, VALID_DESCRIPTION, VALID_TAGS, INVALID_DATE, EMPTY_DATE, EMPTY_DATE);
+        String expectedMessage = Date.MESSAGE_CONSTRAINTS;
+        assertThrows(IllegalValueException.class, expectedMessage, task::toModelType);
+    }
+
+    @Test
+    public void toModelType_invalidFromDate_throwsIllegalValueException() {
+        JsonAdaptedTask task =
+                new JsonAdaptedTask(VALID_NAME, VALID_DESCRIPTION, VALID_TAGS, EMPTY_DATE, INVALID_DATE, EMPTY_DATE);
+        String expectedMessage = Date.MESSAGE_CONSTRAINTS;
+        assertThrows(IllegalValueException.class, expectedMessage, task::toModelType);
+    }
+
+    @Test
+    public void toModelType_invalidToDate_throwsIllegalValueException() {
+        JsonAdaptedTask task =
+                new JsonAdaptedTask(VALID_NAME, VALID_DESCRIPTION, VALID_TAGS, EMPTY_DATE, EMPTY_DATE, VALID_DATE);
+        String expectedMessage = Date.MESSAGE_CONSTRAINTS;
+        assertThrows(IllegalValueException.class, expectedMessage, task::toModelType);
+    }
+
+    @Test
+    public void toModelType_deadlineAndEventOverlap_throwsIllegalValueException() {
+        JsonAdaptedTask task =
+                new JsonAdaptedTask(VALID_NAME, VALID_DESCRIPTION, VALID_TAGS, VALID_DATE, VALID_DATE, VALID_DATE);
+        String expectedMessage = DEADLINE_EVENT_OVERLAP;
+        assertThrows(IllegalValueException.class, expectedMessage, task::toModelType);
+    }
```

Comment:
Nice work. Since we are already testing this case, how about also testing situations where by the JSON date format is  (EMPTY, VALID, EMPTY) and (EMPTY, EMPTY, VALID).